funcbench/main.go

```diff
@@ -172,22 +169,26 @@ func newGitClient(event *github.IssueCommentEvent) (*gitClient, error) {
 		repo:   *event.GetRepo().Name,
 	}
 
-	log.Printf("Git Client: %+v\n", c)
 	return &c, err
 }
 
 func (c *gitClient) cloneRepository() error {
+	if err := os.Chdir(os.Getenv("GITHUB_WORKSPACE")); err != nil {
+		log.Fatalln(err)
+	}
 	_, _, err := execCommand("git", "clone", fmt.Sprintf("https://github.com/%s/%s.git", c.owner, c.repo))
 	if err != nil {
 		return err
 	}
-	err = os.Chdir(c.repo)
-	return err
+	return nil
 }
 
 // checkoutToPullRequest applies changes from the pull request to the working tree
 // of the branch that is being compared.
 func (c *gitClient) checkoutPR(num int) error {
+	if err := os.Chdir(filepath.Join(os.Getenv("GITHUB_WORKSPACE"), c.repo)); err != nil {
+		return errors.Wrap(err, "changing to GITHUB_WORKSPACE dir")
```

Comment:
Personal interest question, is it better to report what is happening when the error occurred or report why the error occurred?

Reply:
I don't quite understand, can you give an example

Reply:
When an error occurs, what is better error description:
"changing to GITHUB_WORKSPACE dir" or
"Could not change directory to XXX"?

Reply:
I think either one works.